src/collection/collectionwatcher.cpp

```diff
@@ -897,6 +952,25 @@ void CollectionWatcher::AddChangedSong(const QString &file, const Song &matching
 
 }
 
+#ifdef HAVE_EBUR128
+void CollectionWatcher::PerformEBUR128Analysis(SongList &songs, ScanTransaction *t) {
+
+  if (!song_ebur128_loudness_analysis_) return;
+
+  // FIXME: this should probably be done on thread pool.
+  for (Song &song : songs) {
+    std::optional<EBUR128Measures> loudness_characteristics = EBUR128Analysis::Compute(song);
+    if (loudness_characteristics) {
+      song.set_ebur128_integrated_loudness_lufs(loudness_characteristics->loudness_lufs);
+      song.set_ebur128_loudness_range_lu(loudness_characteristics->range_lu);
+      if (t)
+        t->new_songs << song;
```

Comment:
The songs can be added twice to new_songs now since they can already be added in AddChangedSong too.

Reply:
Yeah, as i said, i really do not like this commit.
`new_songs` is the right place, but indeed, this can cause duplication.
I'll take a look.

Reply:
There is a number of ways to solve this. I've gone for the most straight-forward one,
but it will obviously be not the most permanent one (since it is `O(n^2)`).
Let me know if this should be improved.

Reply:
Can loudness be different for individual songs for the same file since you do the calculation after UpdateCueAssociatedSongs / UpdateNonCueAssociatedSong?

Reply:
I'm sorry, i think i do not understand the question.

The analysis runs on a single song, it does not matter
where that song is located, is it a mere part of a CUE,
or there is a file per song.